src/mongoose_client_api_rooms.erl

```diff
@@ -150,15 +149,30 @@ determine_role(US, Users) ->
             Role
     end.
 
--spec validate_room_id(RoomIDOrJID :: binary(), Server :: binary()) ->
-    {ok, RoomID :: binary()} | error.
-validate_room_id(RoomIDOrJID, Server) ->
-    MUCLightDomain = muc_light_domain(Server),
+-spec validate_room_id(RoomIDOrJID :: binary() | term(), Server :: binary()) ->
+    {ok, RoomID :: binary(), RoomHost :: binary()} | error.
+validate_room_id(RoomIDOrJID, Server) when is_binary(RoomIDOrJID) ->
     case jid:from_binary(RoomIDOrJID) of
         #jid{luser = <<>>, lserver = RoomID, lresource = <<>>} ->
-            {ok, RoomID};
-        #jid{luser = RoomID, lserver = MUCLightDomain, lresource = <<>>} ->
-            {ok, RoomID};
+            DefaultMucLightDomain = muc_light_domain(Server),
+            {ok, RoomID, DefaultMucLightDomain};
+        #jid{luser = RoomID, lserver = RoomHost, lresource = <<>>} ->
+            case validate_room_host(RoomHost) of
+              ok ->
+                {ok, RoomID, RoomHost};
+              error ->
+                error
+            end;
         _ ->
             error
     end.
+
+-spec validate_room_host(binary()) -> ok | error.
+validate_room_host(RoomHost) ->
+    MucLightDomains = ejabberd_router:dirty_get_routes_to_module(mod_muc_light),
```

Comment:
Feels a bit hackish. Instead of adding this API, can we reuse `mongoose_subhosts:get_host/1`?

Reply:
Still, we're not sure if that subhost is a MUC light domain. IMO router is a more appropriate place to look for that information, because it is exactly the place where that information is registered (I mean the host <-> module mapping).